### PR TITLE
feat(core)!: delete deprecated `preprocessor` & `postprocessor` from `MaskitoOptions`

### LINKED ISSUE #322: 🚀 - Processors should accept array of functions + `maskitoPipe` deletion
### Which package(s) are relevant/related to the feature request?

@maskito/core, @maskito/kit

### Step 1
Replace these lines
https://github.com/Tinkoff/maskito/blob/4c5a7f64b9a8ac209584c75e17ec022674b87c1b/projects/core/src/lib/types/mask-options.ts#L8-L9
with
```ts
 readonly preprocessors?: MaskitoPreprocessor[]; 
 readonly postprocessors?: MaskitoPostprocessor[]; 
```
(plural form + new type)

___

### Step 2

Deprecate this utility
https://github.com/Tinkoff/maskito/blob/4c5a7f64b9a8ac209584c75e17ec022674b87c1b/projects/core/src/lib/utils/pipe.ts#L12-L19
___

### Step 3

All mask generators from `@maskito/kit` should return `Required<MaskitoOptions>`.

___

### Step 4

Update documentation
https://tinkoff.github.io/maskito/core-concepts/processors

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?
```ts
const options: MaskitoOptions = {
    mask: /.../,
    preprocessor: maskitoPipe(
        preprocessor1,
        preprocessor2,
        preprocessor3,
    ),
    postprocessor: postprocessor1,
};
```

Closes #322

## What is the new behavior?
```ts
const options: MaskitoOptions = {
    mask: /.../,
    preprocessors: [
        preprocessor1,
        preprocessor2,
        preprocessor3,
    ],
    postprocessors: [ postprocessor1 ],
};
```

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/core/src/lib/mask.ts [hunk 1]
```diff
@@ -23,15 +23,9 @@ export class Maskito extends MaskHistory {
         ...this.maskitoOptions,
     };
 
-    private readonly preprocessor = maskitoPipe(
-        this.options.preprocessor,
-        ...this.options.preprocessors,
-    );
+    private readonly preprocessor = maskitoPipe(...this.options.preprocessors);
```
Comment (waterplea) [inline projects/core/src/lib/mask.ts:26]: Why spread?

File: projects/core/src/lib/utils/pipe.ts [hunk 2]
```diff
@@ -1,15 +1,13 @@
 import {MaskitoPostprocessor, MaskitoPreprocessor} from '../types';
 
 /**
- * @deprecated Use property `preprocessors` from `MaskitoOptions`
  * @internal
  */
 export function maskitoPipe(
```
Comment (waterplea) [inline projects/core/src/lib/utils/pipe.ts:6]: Should we rename it to just `pipe` now that it is not exported?

File: projects/core/src/lib/utils/transform.ts [hunk 3]
```diff
@@ -17,8 +17,8 @@ export function maskitoTransform(
         ...MASKITO_DEFAULT_OPTIONS,
         ...maskitoOptions,
     };
-    const preprocessor = maskitoPipe(options.preprocessor, ...options.preprocessors);
-    const postprocessor = maskitoPipe(options.postprocessor, ...options.postprocessors);
+    const preprocessor = maskitoPipe(...options.preprocessors);
```
Comment (waterplea) [inline projects/core/src/lib/utils/transform.ts:20]: Unnecessary spread again?

**Review by waterplea — APPROVED** [review]

File: projects/core/src/lib/utils/pipe.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/core/src/lib/utils/pipe.ts:6]: `maskitoPipe` can be useful for unit testing.

For example:
https://github.com/Tinkoff/taiga-ui/blob/673e1344d728a451a6b75f480d698ffae0f82132/projects/addon-commerce/constants/tests/card-holder-mask.spec.ts#L24

File: projects/core/src/lib/mask.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/core/src/lib/mask.ts:26]: Should we change signature of `maskitoPipe` ? 🤔 

```ts
export function maskitoPipe(
    processors: ReadonlyArray<MaskitoPreprocessor | null | undefined>
): MaskitoPreprocessor;

export function maskitoPipe(
    processors: ReadonlyArray<MaskitoPostprocessor | null | undefined>
): MaskitoPostprocessor;
```

What do you think ?

Comment (nsbarsukov) [reply projects/core/src/lib/mask.ts]: Done in https://github.com/Tinkoff/maskito/pull/337/commits/593a9313e56521383520e0db2c07023fe55d1449

Could you review the PR again, please ?

File: projects/core/src/lib/utils/transform.ts (hunk 3 above)
Comment (nsbarsukov) [reply projects/core/src/lib/utils/transform.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**